metaseq/tasks/streaming_language_modeling.py

```diff
@@ -373,24 +380,31 @@ def _tokenize_interleaving_json(self, json):
 
         all_tokens = [int(x) for x in all_tokens]
         final_indexes = torch.LongTensor(all_tokens)
-        return final_indexes
+        return final_indexes, image_spans
 
     def _tokenize_ra_json(self, json):
-        query_index = self.tokenize_single_html_doc(json["text"], add_eod=True)
+        image_spans = []
+        query_index, query_image_span = self.tokenize_single_html_doc(json["text"], add_eod=True)
         query_index = torch.LongTensor(query_index)
         ra_docs = json["retrieved_docs_from_img"] + json["retrieved_docs_from_txt"]
         random.shuffle(ra_docs)
 
         ra_docs = ra_docs[: self.args.num_retrieved_doc]
         ra_indexes = []
+        cur_len = 1
         for ra_doc in ra_docs:
-            ra_index = self.tokenize_single_html_doc(ra_doc, add_eod=False)
+            ra_index, image_span = self.tokenize_single_html_doc(ra_doc, add_eod=False)
             ra_index = torch.LongTensor(ra_index + [self.cm3_break_ind])
             ra_indexes.append(ra_index)
+            image_span = (1 + cur_len + image_span[0], 1 + cur_len + image_span[1])
+            cur_len += len(ra_index)
+            image_spans.append(image_span)
+
         final_indexes = torch.cat(
             [torch.LongTensor([self.eod])] + ra_indexes + [query_index]
         )
-        return final_indexes
+        image_spans.append((1 + cur_len + query_image_span[0], 1 + cur_len + query_image_span[1]))
```

Comment:
Could we add a comment why we're adding 1 (just show we don't have magical numbers)